### PR TITLE
[Merged by Bors] - refactor(GroupTheory/SpecificGroups/Cyclic): Switch from `Fintype` to `Finite`

### PR DESCRIPTION
**Pull request by tb65536**
This PR switches most of `GroupTheory/SpecificGroups/Cyclic.lean` from `Fintype` to `Finite`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 1]
```diff
@@ -311,6 +295,7 @@ theorem IsCyclic.card_pow_eq_one_le [DecidableEq α] [Fintype α] [IsCyclic α]
           exact hm.elim' 1
       simp only [Set.toFinset_card, SetLike.coe_sort_coe]
       rw [Fintype.card_zpowers, orderOf_pow g, orderOf_eq_card_of_forall_mem_zpowers hg]
+      rw [Nat.card_eq_fintype_card]
```
Comment (MichaelStollBayreuth) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:298]: Can this be combined with the preceding line?

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 2]
```diff
@@ -645,29 +629,26 @@ theorem IsCyclic.exponent_eq_zero_of_infinite [Group α] [IsCyclic α] [Infinite
...
   refine ⟨fun h_cyc ↦ (Monoid.exponent_eq_prime_iff hp).mpr fun g hg ↦ ?_, fun h_exp h_cyc ↦ by
-    obtain (rfl|rfl) := eq_zero_or_one_of_sq_eq_self <| hα' ▸ h_exp ▸ (h_cyc.exponent_eq_card).symm
+    obtain (rfl|rfl) := eq_zero_or_one_of_sq_eq_self <| hα ▸ h_exp ▸ (h_cyc.exponent_eq_card).symm
     · exact Nat.not_prime_zero hp
     · exact Nat.not_prime_one hp⟩
   /- we must show every non-identity element has order `p`. By Lagrange's theorem, the only possible
   orders of `g` are `1`, `p`, or `p ^ 2`. It can't be the former because `g ≠ 1`, and it can't
   the latter because the group isn't cyclic. -/
-  have := (Nat.mem_divisors (m := p ^ 2)).mpr ⟨hα' ▸ orderOf_dvd_card (x := g), by aesop⟩
+  have := (Nat.mem_divisors (m := p ^ 2)).mpr ⟨hα ▸ orderOf_dvd_natCard (x := g), by aesop⟩
```
Comment (MichaelStollBayreuth) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:651]: Should the `simp? [...] says` in the next line be removed? IIRC, the `says` combinator re-runs `simp?` each time in any case...

**Comment by urkud** [comment]
!bench

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean (hunk 2 above)
Comment (tb65536) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:651]: I'm not sure what the policy is, but there are plenty of `simp? says` in the library, so if they should be removed, then this ought to be done systematically I feel.

**Comment by tb65536** [comment]
> LGTM. There are still a number of places where one needs to rewrite by `Nat.card_eq_fintype_card` in one direction. Does this suggest that there are some API lemmas missing for one (or both) of the two cardinalities?

Yeah, there were a few places where the proof was annoying enough that I took the coward's way out and just slapped a few `Nat.card_eq_fintype_card`s in the proof (e.g., `IsCyclic.card_pow_eq_one_le` which uses `Fintype` and `Finset`). Eventually these should be switched over as well, but that will take a bit more effort.

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean (hunk 2 above)
Comment (MichaelStollBayreuth) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:651]: OK; I just noticed it here, but I guess this is out of scope for this PR.

**Comment by MichaelStollBayreuth** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**